samples/apps/copilot-chat-app/webapi/CopilotChat/Options/ContentModerationOptions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel.DataAnnotations;
+using SemanticKernel.Service.Options;
+
+namespace SemanticKernel.Service.CopilotChat.Options;
+
+/// <summary>
+/// Configuration options for content moderation.
+/// </summary>
+public class ContentModerationOptions
+{
+    public const string PropertyName = "ContentModeration";
+
+    /// <summary>
+    /// Whether to enable content moderation.
+    /// </summary>
+    [Required, NotEmptyOrWhitespace]
+    public bool Enabled { get; set; } = false;
+
+    /// <summary>
+    /// Azure Content Moderator endpoints
+    /// </summary>
+    [Required, NotEmptyOrWhitespace]
```

Comment:
What happens when content moderation is NOT enabled? I hope we can leave this field and the next empty in that case! Can we?

Reply:
If it's not enabled, then there's a check to disable it from the frontend. Does it matter whether these fields are left empty or not? They don't cause any conflicts if content moderation is disabled, but it helps to have them filled in by default to reduce user friction during setup.

Reply:
I'm specifically asking with regards to appsettings.json.  If they are left empty in that file when moderation is disabled, we don't explode because of the NotEmptyOrWhitespace requirement, do we?

Reply:
Oh I see what you're saying. Hm, not the server starts up successfully:
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/067a9a60-580b-4bd0-ac40-b7e0dd314237)
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/f64cf957-f4b2-4c3a-97c0-4518f5f1d49b)

But I'll change that header to a RequiredOnProperty one

Reply:
Nvm, found out what's happening, it only throws when the service is invoked
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/a26cd1bd-28e0-401f-8ec5-69c25eabc4a5)

I'll leave Endpointed commented out and change the instructions above

Reply:
Changed both to RequiredOnPropertyValue 
![image](https://github.com/microsoft/semantic-kernel/assets/125500434/2b3fd0e0-bfc9-49be-bee1-374981d87db0)

Reply:
Also made the AzureContentModerator service optional